azuredevops/internal/service/git/resource_git_repository_branch.go

```diff
@@ -0,0 +1,239 @@
+package git
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v6/git"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/client"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/converter"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/tfhelper"
+)
+
+// ResourceGitRepositoryBranch schema to manage the lifecycle of a git repository branch
+func ResourceGitRepositoryBranch() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceGitRepositoryBranchCreate,
+		ReadContext:   resourceGitRepositoryBranchRead,
+		DeleteContext: resourceGitRepositoryBranchDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceGitRepositoryBranchImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"repository_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IsUUID,
+			},
+			"ref": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"tag": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"commit_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
```

Comment:
Suggest rename  `ref` -> `ref_branch`,  `tag` -> `ref_tag` , `commit_id` -> `ref_commit_id` and only one of them can be configured at once. `ConflictsWith` can be used to add such constraint.
```suggestion
"ref_branch": {
	Type:          schema.TypeString,
	Optional:      true,
	ForceNew:      true,
	ValidateFunc:  validation.StringIsNotEmpty,
	ConflictsWith: []string{"ref_tag", "ref_commit_id"},
},
"ref_tag": {
	Type:          schema.TypeString,
	Optional:      true,
	ForceNew:      true,
	ValidateFunc:  validation.StringIsNotEmpty,
	ConflictsWith: []string{"ref_tag", "ref_commit_id"},
},
"ref_commit_id": {
	Type:          schema.TypeString,
	Optional:      true,
	ForceNew:      true,
	Computed:      true,
	ValidateFunc:  validation.StringIsNotEmpty,
	ConflictsWith: []string{"ref_tag", "ref_commit_id"},
},
```

Reply:
Ok, I think this is good indeed.